R/XStringSet-io.R

```diff
@@ -365,10 +365,12 @@ readAAStringSet <- function(filepath, format="fasta",
 
 .write_XStringSet_to_fastq <- function(x, filexp_list, qualities=NULL)
 {
-    if (is(x, "QualityScaledXStringSet") && is.null(qualities)) {
-        qualities <- quality(x)
-    } else if (is.null(qualities))
-        qualities <- mcols(x)$qualities
+    if (is.null(qualities)){
+        if (is(x, "QualityScaledXStringSet"))
+            qualities <- quality(x)
+        else
+            qualities <- mcols(x)$qualities
+    }
```

Comment:
this can go straight to an `} else {`.

Reply:
I don't think so. `mcols(x)$qualities` could be `"whatever"` and then it would break. Same is also true if `qualities = "whatever"` is set.

Reply:
Hm. Yes, that's true enough.